code/datums/components/jousting.dm

```diff
@@ -0,0 +1,81 @@
+/datum/component/jousting
+	var/current_direction = NONE
+	var/max_tile_charge = 5
+	var/min_tile_charge = 2				//tiles before this code gets into effect.
+	var/current_tile_charge = 0
+	var/movement_reset_tolerance = 2			//deciseconds
+	var/unmounted_damage_boost_per_tile = 0
+	var/unmounted_knockdown_chance_per_tile = 0
+	var/unmounted_knockdown_time = 0
+	var/mounted_damage_boost_per_tile = 2
+	var/mounted_knockdown_chance_per_tile = 20
+	var/mounted_knockdown_time = 20
+	var/requires_mob_riding = TRUE			//whether this only works if the attacker is riding a mob, rather than anything they can buckle to.
+	var/requires_mount = TRUE				//kinda defeats the point of jousting if you're not mounted but whatever.
+	var/mob/current_holder
+	var/datum/component/redirect/listener
+	var/current_timerid
+
+/datum/component/jousting/Initialize()
+	if(!isitem(parent))
+		. = COMPONENT_INCOMPATIBLE
+		stack_trace("Warning: Jousting component incorrectly applied to invalid parent type [parent.type]")
+	RegisterSignal(COMSIG_ITEM_EQUIPPED, .proc/on_equip)
+	RegisterSignal(COMSIG_ITEM_DROPPED, .proc/on_drop)
+	RegisterSignal(COMSIG_ITEM_ATTACK, .proc/on_attack)
+
+/datum/component/jousting/Destroy()
+	QDEL_NULL(listener)
+	return ..()
+
+/datum/component/jousting/proc/on_equip(mob/user, slot)
+	QDEL_NULL(listener)
+	current_holder = user
+	listener = new(user, COMSIG_MOVABLE_MOVED, CALLBACK(src, .proc/mob_move))
+
+/datum/component/jousting/proc/on_drop(mob/user)
+	QDEL_NULL(listener)
+	current_holder = null
+	current_direction = NONE
+	current_tile_charge = 0
+
+/datum/component/jousting/proc/on_attack(mob/living/target, mob/user)
+	if(user != current_holder)
+		return
+	var/current = current_tile_charge
+	var/obj/item/I = parent
+	var/target_buckled = target.buckled ? TRUE : FALSE			//we don't need the reference of what they're buckled to, just whether they are.
+	if((requires_mount && ((requires_mob_riding && !ismob(user.buckled)) || (!user.buckled))) || !current_direction || (current_tile_charge < min_tile_charge))
+		return
+	var/turf/target_turf = get_step(user, current_direction)
+	if(target in range(1, target_turf))
+		var/knockdown_chance = (target_buckled? mounted_knockdown_chance_per_tile : unmounted_knockdown_chance_per_tile) * current
+		var/knockdown_time = (target_buckled? mounted_knockdown_time : unmounted_knockdown_time)
+		var/damage = (target_buckled? mounted_damage_boost_per_tile : unmounted_damage_boost_per_tile) * current
+		var/msg = ""
+		if(damage)
+			msg += "[user] [I.is_sharp()? "impales" : "slams into"] [target] [I.is_sharp()? "on" : "with"] their [parent]"

```

Comment:
Nitpick, but caching is_sharp() in a var before this line and then using that var instead is (I think!) a tiny bit faster!